src/webgpu/shader/execution/robust_access_vertex.spec.ts

```diff
@@ -44,7 +44,430 @@ Vertex buffer contents could be randomized to prevent the case where a previous
+          gl_VertexIndex < ${vertexIndexOffset + numVertices});
+
+        if (attributesInBounds && (${!p.indexed} || indexInBounds)) {
+          // Success case, move the vertex out of the screen
+          gl_Position = vec4(-1.0, 0.0, 0.0, 1.0);
+        } else {
+          // Failure case, move the vertex inside the screen
+          gl_Position = vec4(0.0, 0.0, 0.0, 1.0);
+        }
+      }
+    `,
+    });
+
+    const fragmentModule = t.makeShaderModule('fragment', {
+      glsl: `
+      #version 450
+      precision mediump float;
+
+      layout(location = 0) out vec4 fragColor;
+
+      void main() {
+        fragColor = vec4(0.0, 1.0, 0.0, 1.0);
+      }
+    `,
+    });
+
+    // Pipeline setup, texture setup
+    const colorAttachment = t.device.createTexture({
+      format: 'rgba8unorm',
+      size: { width: 1, height: 1, depth: 1 },
+      usage: GPUTextureUsage.COPY_SRC | GPUTextureUsage.OUTPUT_ATTACHMENT,
+    });
+    const colorAttachmentView = colorAttachment.createView();
+
+    const pl = t.device.createPipelineLayout({ bindGroupLayouts: [] });
+    const pipeline = t.device.createRenderPipeline({
+      vertexStage: { module: vertexModule, entryPoint: 'main' },
+      fragmentStage: { module: fragmentModule, entryPoint: 'main' },
+      layout: pl,
+      primitiveTopology: 'point-list',
+      rasterizationState: {
+        frontFace: 'ccw',
+      },
+      colorStates: [{ format: 'rgba8unorm', alphaBlend: {}, colorBlend: {} }],
+      vertexState: {
+        indexFormat: 'uint16',
+        vertexBuffers,
+      },
+    });
+
+    // Offset the draw call parameter we are testing by |errorScale|
+    switch (p.drawCallTestParameter) {
+      case DrawCallParameter.VertexCount: {
+        draw.vertexCount += p.errorScale;
+        break;
+      }
+      case DrawCallParameter.FirstVertex: {
+        draw.firstVertex += p.errorScale;
+        break;
+      }
+      case DrawCallParameter.InstanceCount: {
+        draw.instanceCount += p.errorScale;
+        break;
+      }
+      case DrawCallParameter.FirstInstance: {
+        draw.firstInstance += p.errorScale;
+        break;
+      }
+      case DrawCallParameter.IndexCount: {
+        draw.indexCount += p.errorScale;
+        break;
+      }
+      case DrawCallParameter.FirstIndex: {
+        draw.firstIndex += p.errorScale;
+        break;
+      }
+      case DrawCallParameter.BaseVertex: {
+        draw.baseVertex += p.errorScale;
+        break;
+      }
+    }
+
+    const encoder = t.device.createCommandEncoder();
+    const pass = encoder.beginRenderPass({
+      colorAttachments: [
+        {
+          attachment: colorAttachmentView,
+          storeOp: 'store',
+          loadValue: { r: 1.0, g: 0.0, b: 0.0, a: 1.0 },
+        },
+      ],
+    });
+    pass.setPipeline(pipeline);
+
+    // Draw function lookup |drawFunc[indexed][indirect]|
+    const drawFunc: { [k: string]: Function } = {
```

Comment:
Ew for stringly typed stuff.

Reply:
Perhaps nicer to replace this with a method on DrawCall which chooses a draw call based on two booleans.

Reply:
Added a method instead of this